### PR TITLE
Consider constructors in addition to function calls for named arguments

### PR DESCRIPTION
**Pull request by j3parker**
Forgot to handle this. Hopefully the fallout won't be too bad.

### CODE REVIEW

**Comment by j3parker** [comment]
Fallout from missing this isn't too bad.

File: src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs [hunk 1]
```diff
@@ -137,5 +144,20 @@ ArgumentListSyntax args
 				);
 			}
 		}
+
+		// Not an extension method because there may be more cases (e.g. in the
+		// future) and if more than this fix + its analyzer used this logic
+		// there could be undesirable coupling if we handled more cases.
+		internal static ArgumentListSyntax GetArgs( SyntaxNode syntax ) {
+			if ( syntax is InvocationExpressionSyntax ) {
```
Comment (omsmith) [inline src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs]: nit: consider switch with pattern matching?

File: src/D2L.CodeStyle.Analyzers/Language/UseNamedArgumentsCodeFix.cs [hunk 2]
```diff
@@ -120,5 +114,21 @@ CancellationToken cancellationToken
...
 			}
 		}
+
+		public static ArgumentListSyntax GetArgs(
+			SyntaxNode root,
+			TextSpan span
+		) {
+			// FindNode() may identify a parent of the ExpressionSyntax
+			// we'd like to fix that has an equal span. We're grabbing the
+			// "first" child InvocationExpressionSyntax. I'm guessing that will
```
Comment (omsmith) [inline src/D2L.CodeStyle.Analyzers/Language/UseNamedArgumentsCodeFix.cs]: nit: comment references invocationexpression

File: src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs [hunk 3]
```diff
@@ -137,5 +144,20 @@ ArgumentListSyntax args
 				);
 			}
 		}
+
+		// Not an extension method because there may be more cases (e.g. in the
+		// future) and if more than this fix + its analyzer used this logic
+		// there could be undesirable coupling if we handled more cases.
+		internal static ArgumentListSyntax GetArgs( SyntaxNode syntax ) {
+			if ( syntax is InvocationExpressionSyntax invocation ) {
```
Comment (omsmith) [inline src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs]: That's cool, but I meant

```csharp
switch( syntax ) {
  case InvocationExpressionSyntax invocation:
    return invocation.ArgumentList;
  case ObjectCreationExpressionSyntax objectCreation:
    return objectCreation.ArgumentList;
  default:
    return null;
}
```

Comment (j3parker) [reply src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs]: Whoa cool tech

(I wish you could have sealed enum types so `default` wasn't necessary.)

**Comment by j3parker** [comment]
That last commit is the right way to use `FindNode`. The thing I was doing with `.DescendantNodes` was wrong.

**Comment by j3parker** [comment]
Passes in the LMS once some stuff gets merged. Will wait a couple of days after merging to deploy to maybe avoid aggro due to folks checkouts.

**Comment by j3parker** [comment]
I think I've waited long enough for this. I don't think waiting really would help much but wanted to space out the inevitable split-checkout breakages that people have a bit.

**Pull request merged**